File: chisel_py/src/chisel/checker/models/agent_skill.py
from dataclasses import dataclass
from enum import Enum


class SkillTarget(str, Enum):
    CODEX = "codex"
    CLAUDE = "claude"
    OPENCODE = "opencode"


@dataclass(frozen=True, slots=True)
class SkillInstallResult:
    name: str
    source: str
    destination: str
    status: str
    reason: str


@dataclass(frozen=True, slots=True)
class SkillSetupResult:
    target: SkillTarget
    target_dir: str
    results: list[SkillInstallResult]

File: chisel_js/src/chisel/checker/models/agent_skill.ts
export enum SkillTarget {
  CODEX = "codex",
  CLAUDE = "claude",
  OPENCODE = "opencode",
}

export interface SkillInstallResult {
  readonly name: string;
  readonly source: string;
  readonly destination: string;
  readonly status: string;
  readonly reason: string;
}

export interface SkillSetupResult {
  readonly target: SkillTarget;
  readonly targetDir: string;
  readonly results: SkillInstallResult[];
}
